sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorStateStore.java

```diff
@@ -193,17 +192,27 @@ public void storeTasks(Collection<Protos.TaskInfo> tasks) throws StateStoreExcep
 
     @Override
     public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
-        String taskName;
-        try {
-            taskName = CommonTaskUtils.toTaskName(status.getTaskId());
-        } catch (TaskException e) {
-            throw new StateStoreException(Reason.LOGIC_ERROR, String.format(
-                    "Failed to parse the Task Name from TaskStatus.task_id: '%s'", status), e);
+        Optional<String> optionalTaskName = Optional.empty();
+        for (Protos.TaskInfo taskInfo : fetchTasks()) {
+            if (taskInfo.getTaskId().getValue().equals(status.getTaskId().getValue())) {
+                if (optionalTaskName.isPresent()) {
+                    logger.error("Found duplicate taskIDs in Task {} and  Task {}",
+                            optionalTaskName.get(), taskInfo.getName());
+                    throw new StateStoreException(Reason.LOGIC_ERROR, String.format(
+                            "There are more than one tasks with TaskID: %s", status));
+                }
+                    optionalTaskName = Optional.of(taskInfo.getName());
+            }
         }
+        if (!optionalTaskName.isPresent()) {
+            throw new StateStoreException(Reason.NOT_FOUND, String.format(
```

Comment:
This is an undesirable behavior change.  See the comment from the old code.  It is a pretty common occurrence to receive stale TaskStatus messages.  Logging a warning would be fine, but throwing an exception isn't a good idea.